core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -481,12 +487,49 @@ public static void deleteCancerStudy(String cancerStudyStableId) throws DaoExcep
             JdbcUtil.closeAll(DaoCancerStudy.class, con, pstmt, rs);
         }
     }
+    
+    /**
+     * Calls deleteCancerStudyByCascade() if cancer study exists by stable id.
+     * @param cancerStudyStableId
+     * @throws DaoException 
+     */
+    public static void deleteCancerStudyByCascade(String cancerStudyStableId) throws DaoException {
+        CancerStudy study = getCancerStudyByStableId(cancerStudyStableId);
+        if (study != null) {
+            deleteCancerStudyByCascade(study.getInternalId());
+        }
+    }
 
     /**
+     * Deletes a cancer study by internal id from db with foreign key constraints enforced.
+     * @param internalCancerStudyId
+     * @throws DaoException 
+     */
+    static void deleteCancerStudyByCascade(int internalCancerStudyId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            pstmt = con.prepareStatement("DELETE FROM cancer_study WHERE cancer_study_id = ?");
+            pstmt.setInt(1, internalCancerStudyId);
+            pstmt.executeUpdate();
+            removeCancerStudyFromCache(internalCancerStudyId);
```

Comment:
do we need both an explicit removeCancerStudyFromCache(id), and also the more general reCacheAll() below? I'm not familiar with reCacheAll .. but if it deletes the whole cache and then recaches maybe this is not needed here?